tests/shotgun_model/test_data_handler_find.py

```diff
@@ -0,0 +1,247 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sys
+import os
+
+from mock import patch, Mock, call
+from tank_test.tank_test_base import *
+
+# import the test base class
+test_python_path = os.path.abspath(os.path.join( os.path.dirname(__file__), "..", "python"))
+sys.path.append(test_python_path)
+from base_test import TestShotgunUtilsFramework
+
+
+class TestShotgunFindDataHandler(TestShotgunUtilsFramework):
+    """
+    Tests for the data handler low level io
+    """
+    
+    def setUp(self):
+        """
+        Fixtures setup
+        """        
+        super(TestShotgunFindDataHandler, self).setUp()
+        self.shotgun_model = self.framework.import_module("shotgun_model")
+
+    def test_generate_data_request(self):
+        """
+        Test generate_data_request
+        """
+        test_path = os.path.join(self.tank_temp, "test_find_handler.pickle")
+
+        dh = self.shotgun_model.data_handler_find.ShotgunFindDataHandler(
+            entity_type="Asset",
+            filters=[],
+            order=None,
+            hierarchy=["sg_asset_type", "code"],
+            fields=["code"],
+            download_thumbs=True,
+            limit=None,
+            additional_filter_presets=None,
+            cache_path=test_path,
+            parent=None
+        )
+
+        dh.load_cache()
```

Comment:
Nitpick: All your tests start with

```python
dh = ShotgunFinDataHandler(...)
dg.load_cache()
```

To me that's a sign that maybe the cache should be lazy loaded by update operations. Otherwise the API is error prone. It's an internal API with very limited scope, so it's not that big of a deal. Curious what you think.